HashMap.java

```diff
@@ -0,0 +1,101 @@
+public class HashMap<E,E1> {
+	
+	private E [] key ;
+	private E1 [] value;
+	int size = 10000;
+	
+//	private static int index = -1;
+	
+	@SuppressWarnings("unchecked")
+	public HashMap()
+	{
+		key = (E[]) new Object[size];
+		value = (E1[]) new Object[size];
+	}
+	
+
+	public E1 get(E key)
+	{
+		int index = getIndexValue(key);
+		int count = 0;
+		
+	
+		if(this.key[index] == key)
+			return (E1)this.value[index];
+		
+		else{
+			while(this.key[index] != key)
+			{
+				index++;count++;
+			
+				if(index == size)
+					index = 0;
+				if(count == size - 1)
+					return null;
+			}
+		
+			return (E1)this.value[index];
+		}
+		
+	}
+	
+
+	public void put(E key, E1 value)
+	{
+		int index = getIndexValue(key);
+		
+		if(get(key) == null){
+			if(this.key[index] == null){
+				
+				this.key[index] = key;
+				this.value[index] = value;
+			}
+			else{
+				while(this.key[index] != null)
+				{
+					index++;
+					if(index == size)
+						index = 0;
+				}
+			
+				this.key[index] = key;
+				this.value[index] = value;
+			}
+		}	
+		else System.out.println("Key already exist.");
+	}
+	
+	public int getIndexValue(E key)
+	{
+		int index = 0;
+		
+		if(key instanceof Integer){
+			//System.out.println("Integer " + key);

```

Comment:
How this time can be saved? How we can move this comparison to compile time so that compiler can optimise this?